Makefile

```diff
@@ -132,29 +132,35 @@ define n
 
 endef
 $(info $(subst CC,$nCC,$(shell $(MAKE) VERBOSE=$(VERBOSE) libfreeradius-make-dlopen.${BUILD_LIB_EXT} libfreeradius-make-version.${BUILD_LIB_EXT})))
-endif
+endif # BUILD_MAKE_LIBS
 
 load build/lib/.libs/libfreeradius-make-dlopen.${BUILD_LIB_EXT}(dlopen_gmk_setup)
 load build/lib/.libs/libfreeradius-make-version.${BUILD_LIB_EXT}(version_gmk_setup)
 
-else
+else # libfreeradius-make
 #
 #  We're building ONLY the libfreeradius-make-* files.
 #  Leave the outputs at the default location, but take the
 #  inputs from the scripts/build directory.
 #
 BUILD_DIR:=${top_srcdir}/build
 top_builddir:=${top_srcdir}/scripts/build
-endif
-endif
+endif # libfreeradius-make
+endif # clean
 
 #
 #  Load the huge boilermake framework.
 #
 include scripts/boiler.mk
-endif
-endif
-endif
+endif # crossbuild
+else # rpm
+#
+#  We're building rpm
+#
+top_srcdir=	$(realpath $(dir $(lastword $(MAKEFILE_LIST))))
+BUILD_DIR=	build
```

Comment:
This would be a bit clearer if it were in a separate `ifeq "$(MAKECMDGOALS)" "rpm"` section after the endifs, rather than in an else. This if section is rather large already.

Reply:
BUILD_DIR and top_srcdir is defined for the other case (libfreeradius-make) in the same block